### PR TITLE
Uptake cert-manager version being used by Verrazzano from 0.13 to 1.2.0

### PR DESCRIPTION
**Pull request by speidelj**
# Description

Change the version of cert-manager that is being used by Verrazzano. Previous version was 0.13 and new version is 1.2.0.

Fixes VZ-1873

Important note: Previously we had cert-manager configured to disable the webhook and cainjector components of cert-manager even though this was strongly discouraged in the cert-manager documentation. In the 1.2.0 version that we are upgrading to, it doesn't appear that webhooks can be disabled. 

# Checklist 

As the author of this PR, I have:

- [ x] Checked that I included or updated copyright and license notices in all files that I altered
- [ x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Comment by speidelj** [comment]
I still need to rebase against master but wanted to get the reviews started.

File: go.mod [hunk 1]
```diff
@@ -10,11 +10,12 @@ require (
 	github.com/crossplane/crossplane-runtime v0.10.0
 	github.com/crossplane/oam-kubernetes-runtime v0.3.2
 	github.com/gertd/go-pluralize v0.1.7
-	github.com/go-logr/logr v0.1.0
+	github.com/go-logr/logr v0.4.0
```
Comment (bobdonat) [inline go.mod:13]: is this an approved version

File: go.mod [hunk 2]
```diff
@@ -27,18 +28,13 @@ require (
 	k8s.io/apiextensions-apiserver v0.19.0
 	k8s.io/apimachinery v0.19.0
 	k8s.io/client-go v12.0.0+incompatible
-	k8s.io/code-generator v0.18.6
+	k8s.io/code-generator v0.19.0
```
Comment (bobdonat) [inline go.mod:31]: same comment, is this an approved version

File: platform-operator/helm_config/overrides/cert-manager-values.yaml [hunk 3]
```diff
@@ -3,18 +3,11 @@
...
   repository: ghcr.io/verrazzano/cert-manager-controller
-  tag: 0.13.1-20201016205232-4c8f3fe38
-
-webhook:
-  enabled: false
-  injectAPIServerCA: false
-
-cainjector:
-  enabled: false
+  tag: 1.2.0-20210510184928-aac6bdf62
```
Comment (bobdonat) [inline platform-operator/helm_config/overrides/cert-manager-values.yaml:6]: this stuff is going to conflict with the PR from Fred/Mike C/Tony

File: go.mod (hunk 1 above)
Comment (speidelj) [reply go.mod:13]: These version changes were discussed with @markxnelson previously. They were required due to the new cert-manager version using a newer version of a library which changed the name of an api. The go.mod changes were the smallest possible change that would work with the new cert-manager version.

File: go.mod (hunk 2 above)
Comment (speidelj) [reply go.mod:31]: see above comment

File: platform-operator/helm_config/overrides/cert-manager-values.yaml (hunk 3 above)
Comment (speidelj) [reply platform-operator/helm_config/overrides/cert-manager-values.yaml:6]: yeah, although my changes here are pretty simple and I wouldn't expect a merge to be difficult in this file

**Review by markxnelson — APPROVED** [review]

**Pull request merged**